test_runner/regress/test_compaction.py

```diff
@@ -217,6 +226,16 @@ def test_sharding_compaction(
     # Assert that everything is still readable
     workload.validate()
 
+    if gc_compaction:
+        for shard in env.storage_controller.locate(tenant_id):
+            pageserver = env.get_pageserver(shard["node_id"])
+            tenant_shard_id = shard["shard_id"]
+            pageserver.http_client().timeline_compact(
+                tenant_shard_id,
+                timeline_id,
+                enhanced_gc_bottom_most_compaction=True,
+            )
```

Comment:
A code comment on why we run the gc compaction here is essential for future readers (and me :) )

I presume you're piggybacking on this test to get some more coverage?
Also, could my `timeline_scan_no_disposable_keys` help you in this or other tests?

Reply:
yes, that's really helpful and I'll add that in the next patch :) Added a comment to explain why the gc-compaction is triggered there.